projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-replenishment-order-history.ts

```diff
@@ -0,0 +1,137 @@
+
+export function waitForReplenishmentRequest(requestMethod: string) {
+  const replenishmentAlias = 'replenishmentAlias';
+
+  cy.server();
+
+  cy.route(
+    requestMethod,
+    `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
+      'BASE_SITE'
+    )}/users/*/replenishmentOrders*`
+  ).as(replenishmentAlias);
+
+  return `@${replenishmentAlias}`;
+}
+
+export function visitReplenishmentHistory() {
+  const replenishmentHistory = waitForReplenishmentRequest('GET');
+
+  cy.visit(replenishmentOrderHistoryUrl);
+  cy.wait(replenishmentHistory).its('status').should('eq', 200);
+
+  return replenishmentHistory;
+}
+
+export function waitForReplenishmentOrders() {
+  doPlaceOrder(b2bProduct).then(() => {
+    doPlaceOrder(b2bProduct).then((orderData: any) => {
+      cy.waitForOrderToBePlacedRequest(
+        POWERTOOLS_BASESITE,
+        undefined,
+        orderData.body.replenishmentOrderCode
+      );
+
+      const replenishmentHistory = visitReplenishmentHistory();
+
+      cy.get(replenishmentHistory).should((xhr) => {
+        const body = xhr.response.body;
+
+        expect(body.replenishmentOrders).to.have.length(2);
+
+        body.replenishmentOrders.forEach((replenishmentOrder) => {
+          expect(replenishmentOrder.active).to.equal(true);
+        });
+      });
+
+      cy.get(
+        `${replenishmentOrderHistorySelector} .cx-replenishment-order-history-header h3`
+      ).should('contain', 'Replenishment Order History');
+      cy.get(
+        `${replenishmentOrderHistorySelector} .cx-replenishment-order-history-code > .cx-replenishment-order-history-value`
+      ).should('contain', orderData.body.replenishmentOrderCode);
+      cy.get(
+        `${replenishmentOrderHistorySelector} .cx-replenishment-order-history-total > .cx-replenishment-order-history-value`
+      ).should('contain', orderData.body.totalPriceWithTax.formattedValue);
+    });
+  });
+}
+
+export function verifySorting() {
+  visitReplenishmentHistory();
+
+  cy.server();
+  cy.route('GET', /sort=byReplenishmentNumber/).as('query_order_asc');
+
+  cy.get('.top cx-sorting .ng-select').ngSelect('Replenishment Number');
+
+  cy.wait('@query_order_asc').its('status').should('eq', 200);
+
+  cy.get(
+    `${replenishmentOrderHistorySelector} .cx-replenishment-order-history-code > .cx-replenishment-order-history-value`
+  ).then(($orders) => {
+    expect(
+      $orders[0].textContent
+        .toString()
+        .localeCompare($orders[1].textContent.toString())
+    ).to.equal(-1);
+  });
+}
+
+export function cancelReplenishmentInHistory() {
+  const replenishmentHistory = visitReplenishmentHistory();
+
+  cy.get(replenishmentHistory).should((xhr) => {
+    const body = xhr.response.body;
+    expect(body.replenishmentOrders[0].active).to.equal(true);
+  });
+
+  verifyFirstRowReplenishmentIsNotCancelled();
+  cy.get(
+    `${replenishmentOrderHistorySelector} .cx-replenishment-order-history-cancel:first button`
+  )
+    .should('exist')
```

Comment:
I think this is good because we want it to fail if it really does not exist. The test starts with 2 placed orders which aren't cancelled therefore always being available at first, so this makes sure the button still exist in the row before clicking it.

But, this is added assertion is explicit vs implicitly stating it. So cy.get('any').should('exist') is the same as doing cy.get('any'), but I wanted to explicitly state that it should exist. I can remove if we want to go with the implicit way.


we can discuss if you'd like